h2o-r/h2o-package/R/predict.R

```diff
@@ -180,17 +180,16 @@ h2o.mojo_predict_csv <- function(input_csv_path, mojo_zip_path, output_csv_path=
 #' @return Returns a data.frame containing computed predictions
 #' @export
 h2o.mojo_predict_df <- function(frame, mojo_zip_path, genmodel_jar_path=NULL, classpath=NULL, java_options=NULL, verbose=F, setInvNumNA=F) {
-	tmp_dir <- tempdir()
-	dir.create(tmp_dir)
+	input_csv_path <- file.path(tempdir(), 'input.csv')
+	prediction_csv_path <- file.path(tempdir(), 'prediction.csv')
```

Comment:
can we create a sub-directory in the temp directory and remove that one instead? this seems prone to errors

Reply:
It's the same thing with the subdirectory. It can be overridden by one, unless having a unique name.

The right solution seems to be generating those filenames with a portion of randomness. Luckily, H2O has transitive dependency on the `stringi` library, which makes it easy to do.

Reply:
I dont think using a transitive dependency is correct, you need to add it as an direct dependcy when used in the code. however I guess to venture that you dont need an external lib to generate a random number in R

Reply:
Letters are used instead of base 10, as the string is then shorter for the same number of possible combinations (there are more letters per place & only 10 numbers). But I guess numbers are fine as well in this case.

runif() is able to quickly produce pseudo-random numbers with uniform distribution once the right arguments are put inside (source: https://renenyffenegger.ch/notes/development/languages/R/functions/runif ). Ran a test myself, the histogram on 10k numbers is nearly uniform, which is good enough for me.